lib/unifex/code_generator/tie_header.ex

```diff
@@ -0,0 +1,32 @@
+defmodule Unifex.CodeGenerator.TieHeader do
+  @moduledoc """
+  Generates connective header file that includes proper header based
+  on selected interface.
+  """
+
+  import Unifex.CodeGenerator.Utils, only: [sigil_g: 2]
+
+  def generate_tie_header(name, interfaces) do
+    ~g"""
+    #pragma once
+
+    #{generate_includes(name, interfaces)}
+    """
+  end
+
+  defp generate_includes(name, interfaces) do
+    interfaces
+    |> Enum.map(&generate_include(name, &1))
+    |> Enum.join("\n")
+  end
+
+  defp generate_include(name, interface) do
+    interface = Unifex.Helper.get_module_string(interface)
+
+    ~g"""
+    #ifdef BUNDLEX_#{interface |> String.upcase()}
```

Comment:
I would make a separate callback in the CodeGenerator for that and implement in the code generators like
```
defmodule Unifex.CodeGenerators.CNode do
...
@impl true
def identification_constant(), do: "BUNDLEX_CNODE"
```